.github/workflows/cmake.yml

```diff
@@ -0,0 +1,190 @@
+name: CMake Build Matrix
+
```

Comment:
@uwerat this is a proposal for a build matrix realizied with a cross product of configs x steps

PS: should we disable this and your CI workflow in order to reduce costs while WIP?